### PR TITLE
Move remote debug buttons to a single menu.

### PR DESCRIPTION
**Pull request by bruvzg**
With https://github.com/godotengine/godot/pull/63312 and https://github.com/godotengine/godot/pull/70662 editor toolbar is getting too cluttered (if multiple remote debug options are enabled), so it makes sense to group all remote debug items in a single menu button.

Before:
icons

After:
menu

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]
Looks great!

File: editor/editor_run_native.cpp [hunk 1]
```diff
@@ -38,81 +38,59 @@
...
+			remote_debug->set_icon(get_theme_icon(SNAME("PlayRemote"), SNAME("EditorIcons")));
+			remote_debug->set_tooltip_text(TTR("Remote Debug"));
+			add_child(remote_debug);
+		} break;
+		case NOTIFICATION_THEME_CHANGED: {
+			if (remote_debug) {
+				remote_debug->set_icon(get_theme_icon(SNAME("PlayRemote"), SNAME("EditorIcons")));
 			}
 		} break;
-
```
Comment (akien-mga) [inline editor/editor_run_native.cpp:64]: Tiny nitpick, but the empty line between `case`s in `_notification` switches is intentional. Those `case`s are often pretty big, function-like blocks so have one extra line between those blocks helps with readability (notably to read the `NOTIFICATION_*` value after `case` in my opinion.

File: editor/editor_run_native.cpp [hunk 2]
```diff
@@ -38,52 +38,36 @@
...
-						mb->set_icon(small_icon);
-						add_child(mb);
-						menus[i] = mb;
-					}
-				}
+			remote_debug = memnew(MenuButton);
+			remote_debug->get_popup()->connect("id_pressed", callable_mp(this, &EditorRunNative::run_native));
+			remote_debug->set_icon(get_theme_icon(SNAME("PlayRemote"), SNAME("EditorIcons")));
+			remote_debug->set_tooltip_text(TTR("Remote Debug"));
+			add_child(remote_debug);
```
Comment (YuriSizov) [inline editor/editor_run_native.cpp:45]: Any particular reason we don't do it in the constructor? Seems like we add it unconditionally anyway. Would also remove the need for a conditional below, on theme changes.

File: editor/editor_run_native.cpp [hunk 3]
```diff
@@ -94,25 +78,21 @@ void EditorRunNative::_notification(int p_what) {
...
 
-	if (p_idx == -1) {
-		if (eep->get_options_count() == 1) {
-			menus[p_platform]->get_popup()->hide();
-			p_idx = 0;
-		} else {
-			return ERR_INVALID_PARAMETER;
-		}
+	if (!EditorNode::get_singleton()->ensure_main_scene(true)) {
+		resume_id = p_id;
```
Comment (YuriSizov) [inline editor/editor_run_native.cpp:90]: Shouldn't we return early here with `OK` if there is no main scene?

File: editor/editor_run_native.cpp (hunk 2 above)
Comment (bruvzg) [reply editor/editor_run_native.cpp]: Moved to constructor.

**Comment by akien-mga** [comment]
Thanks!

**Comment by Deozaan** [comment]
This is confusing when no remote debugs are configured. The button turns blue when clicked, but nothing else happens. Other nearby buttons are disabled or give a warning when you click them but something isn't configured properly.

**Pull request merged**